plugin/src/main/java/org/autorefactor/refactoring/rules/WakeLockRefactoring.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    private Statement createWakelockReleaseNode(MethodInvocation methodInvocation) {
+        final ASTBuilder b = this.ctx.getASTBuilder();
+        IfStatement ifStatement = b.getAST().newIfStatement();
+        // test-clause
+        PrefixExpression prefixExpression = b.getAST().newPrefixExpression();
+        prefixExpression.setOperator(PrefixExpression.Operator.NOT);
+        MethodInvocation isHeldInvocation = b.getAST().newMethodInvocation();
+        isHeldInvocation.setName(b.simpleName("isHeld"));
+        isHeldInvocation.setExpression(b.copyExpression(methodInvocation));
+        prefixExpression.setOperand(isHeldInvocation);
+        ifStatement.setExpression(prefixExpression);
+        // then
+        MethodInvocation releaseInvocation = b.getAST().newMethodInvocation();
+        releaseInvocation.setName(b.simpleName("release"));
+        releaseInvocation.setExpression(b.copyExpression(methodInvocation));
+        ExpressionStatement releaseExpressionStatement = b.getAST().newExpressionStatement(releaseInvocation);
+        ifStatement.setThenStatement(b.block(releaseExpressionStatement));
+        return ifStatement;
+    }
+
+    private MethodDeclaration createOnPauseDeclaration() {
+        final ASTBuilder b = this.ctx.getASTBuilder();
+        MethodDeclaration onPauseDeclaration = b.getAST().newMethodDeclaration();
+        onPauseDeclaration.setName(b.simpleName("onPause"));
+        //
+        NormalAnnotation annotation = b.getAST().newNormalAnnotation();
+        annotation.setTypeName(b.name("Override"));
+        onPauseDeclaration.modifiers().add(annotation);
+        //
+        Modifier protectedModifier = b.getAST().newModifier(ModifierKeyword.PROTECTED_KEYWORD);
+        onPauseDeclaration.modifiers().add(protectedModifier);
+        //
+        SuperMethodInvocation superMethodInvocation = b.getAST().newSuperMethodInvocation();
+        superMethodInvocation.setName(b.simpleName("onPause"));
+
+        //
+        ASTRewrite rewriter = ASTRewrite.create(b.getAST());
+        Block block = b.block(
+                b.newlinePlaceholder(),
+                b.getAST().newExpressionStatement(superMethodInvocation),
+                b.newlinePlaceholder());
+        onPauseDeclaration.setBody(block);
+        return onPauseDeclaration;
+    }
+
+    private MethodDeclaration findMethodOfType(String methodToFind,
+            TypeDeclaration typeDeclaration) {
+        MethodDeclaration[] methods = typeDeclaration.getMethods();
+
+        for (MethodDeclaration method : methods) {
+            IMethodBinding methodBinding = method.resolveBinding();
+            if (methodBinding != null
+                    && methodToFind.equals(methodBinding.getName())) {
+                return method;
+            }
+        }
+        return null;
+    }
+
+
+    private class ReleasePresenceChecker extends ASTVisitor {
+        private boolean releasePresent = false;
+        public boolean isReleasePresent() {
```

Comment:
Pleae inline this method since this in ner class is local to this refactoring rule.